### ISSUE TITLE
msi installer doesn't detect old versions of clamAV installed under a non-standard pathname.

### ISSUE THREAD

**Issue by fernandomt08** [Cisco-Talos/clamav#861]
We are trying to update multiple servers with the latest version of clamav and we are using the msi installer that is provided for Windows systems.
The installer updates the software files if they were previously under the standard pathname C:\Program Files\ClamAV. However, if the software was installed with a custom pathname(i.e. D:\Software), the installer will deploy the files to C:\Program Files\ClamAV by default.

Is there any option with msi installer that allows to specific the installation directory? or perhaps a parameter that forces the installer to lookup for a current installation on the system?

**Comment by micahsnyder** [comment]
The ClamAV MSI installer will remember your install path for upgrading for a specific feature version (e.g. 0.105, 1.0, 1.1, etc). 

The installer stores the path in `HKEY_LOCAL_MACHINE\Software\Kitware\CMake\Packages\ClamAV` in a registry key using this upgrade GUID:
https://github.com/Cisco-Talos/clamav/blob/719a8b2feacbbb7e91ed57805640cd450a58ab22/CMakeLists.txt#L348

To test this, I installed:
0.105.1 to `C:\Program Files\ClamAV`
1.0.0 to `C:\ClamAV`

In the registry, I see:
![image](https://user-images.githubusercontent.com/30635813/225468422-7551ff3e-d0f1-4ce3-bb4e-7fce01916458.png)

The key named `D9F136C1-3691-47E3-9079-4FE9C9006900` stores the path where I installed 0.105.1.
The key named `D9F136C1-3691-47E3-9079-4FE9C9010000` stores the path where I installed 1.0.0.

If I go to install 0.105.2, it will try to upgrade the installation at `C:\Program Files\ClamAV`
And if I go to install 1.0.1, it will try to upgrade the installation at `C:\ClamAV`

If we changed that line of code from:
```cmake
SET(CPACK_WIX_UPGRADE_GUID           "D9F136C1-3691-47E3-9079-4FE9C9${HEX_MAJOR}${HEX_MINOR}00")
```
to:
```cmake
SET(CPACK_WIX_UPGRADE_GUID           "D9F136C1-3691-47E3-9079-4FE9C9${HEX_MAJOR}0000")
```
Then it would be possible for the installer to remember the install path for upgrading across all 1.* versions (until we hit 2.0).  But I'm not sure if this is desirable.  For anyone who wants to have multiple clamav feature versions installed, they would no longer be able to use the MSI installer.  To test, I tried installing 1.0.0 to `C:\ClamAV-1.0.0` and then tried installing 1.0.1 to `C:\ClamAV-1.0.1`.  The result was that the 1.0.0 version was removed and when the 1.0.1 was installed, even though it was to a different path.
So for those people, they would have to use the ZIP instead of the MSI.  

Granted, I don't actually know if anyone uses multiple Clam installs on Windows with the MSI installer.  We don't.  Meanwhile, your use case would be fairly common.
